src/main/java/org/spongepowered/api/text/TextTemplate.java

```diff
@@ -0,0 +1,473 @@
+     * elements is the order in which they will be appended to the result
+     * builder via {@link #apply(Map)}.
+     *
+     * <p>The provided elements may be of any type.</p>
+     *
+     * <p>In the case that an element is a {@link TextElement},
+     * {@link TextElement#applyTo(Text.Builder)} will be used to append the
+     * element to the builder.</p>
+     *
+     * <p>In the case that an element is an {@link Arg} the argument will be
+     * replaced with the {@link TextElement} provided by the corresponding
+     * parameter supplied by {@link #apply(Map)}</p>
+     *
+     * <p>In the case that an element is any other type, the parameter value's
+     * {@link Object#toString()} method will be used to create a {@link Text}
+     * object.</p>
+     *
+     * @param elements Elements to append to builder
+     * @param openArg String to use for beginning of Arg containers
+     * @param closeArg String to use for end of Arg containers
+     * @return Newly constructed TextTemplate
+     */
+    public static TextTemplate of(String openArg, String closeArg, Object[] elements) {
+        checkNotNull(elements, "elements");
+        if (elements.length == 0) {
+            return of();
+        }
+        return new TextTemplate(openArg, closeArg, elements);
+    }
+
+    /**
+     * Constructs a new TextTemplate for the given elements. The order of the
+     * elements is the order in which they will be appended to the result
+     * builder via {@link #apply(Map)}.
+     *
+     * <p>The provided elements may be of any type.</p>
+     *
+     * <p>In the case that an element is a {@link TextElement},
+     * {@link TextElement#applyTo(Text.Builder)} will be used to append the
+     * element to the builder.</p>
+     *
+     * <p>In the case that an element is an {@link Arg} the argument will be
+     * replaced with the {@link TextElement} provided by the corresponding
+     * parameter supplied by {@link #apply(Map)}</p>
+     *
+     * <p>In the case that an element is any other type, the parameter value's
+     * {@link Object#toString()} method will be used to create a {@link Text}
+     * object.</p>
+     *
+     * @param elements Elements to append to builder
+     * @return Newly constructed TextTemplate
+     */
+    public static TextTemplate of(Object... elements) {
+        return of(DEFAULT_OPEN_ARG, DEFAULT_CLOSE_ARG, elements);
+    }
+
+    /**
+     * Returns the empty representation of a TextTemplate.
+     *
+     * @return Empty TextTemplate
+     */
+    public static TextTemplate of() {
+        return EMPTY;
+    }
+
+    /**
+     * Constructs a new {@link Arg} to be supplied to {@link #of(Object...)}.
+     * This argument expects a {@link TextElement} parameter.
+     *
+     * @param name name of argument
+     * @return argument builder
+     */
+    public static Arg.Builder arg(String name) {
+        return new Arg.Builder(name);
+    }
+
+    @Override
+    public Text toText() {
```

Comment:
Is it possible to have a `toText(Text.Builder)` method? It'd be nice to pass an already-made builder here, if available